lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -372,6 +382,141 @@ function processBucketPolicy(requestType, bucket, canonicalID, arn, bucketOwner,
     return processedResult;
 }
 
+/**
+ *
+ * @param {BucketInfo} bucket - bucket object
+ * @param {Account} account - account object
+ * @param {array} apiNames - action names: operations to authorize
+ * @param {string} apiMethod - the main API call
+ * @param {number} inflight - inflight bytes
+ * @param {Logger} log - logger
+ * @param {function} callback - callback function
+ * @returns {boolean} - true if the quota is valid, false otherwise
+ */
+function validateQuotas(bucket, account, apiNames, apiMethod, inflight, log, callback) {
+    const bucketQuota = bucket.getQuota();
+    const accountQuota = account?.quota || 0;
+    let bucketQuotaExceeded = false;
+    let accountQuotaExceeded = false;
+
+    if ((bucketQuota <= 0 && accountQuota <= 0) || !ScubaClientInstance?.enabled) {
+        if (bucketQuota > 0 || accountQuota > 0) {
+            log.warn('quota is set for a bucket, but scuba is disabled', {
+                bucketName: bucket.getName(),
+            });
+        }
+        return callback(null);
+    }
+    const creationDate = new Date(bucket.getCreationDate()).getTime();
+
+    return async.forEach(apiNames, (apiName, done) => {
+        let shouldEvaluateCopyObject = false;
+        if (actionNeedQuotaCheckCopy(apiName,  apiMethod)) {
+            shouldEvaluateCopyObject = true;
+            // eslint-disable-next-line no-param-reassign
+            inflight = Math.abs(inflight);
+        }
+        if (!shouldEvaluateCopyObject && !actionNeedQuotaCheck[apiName] && !actionWithDataDeletion[apiName]) {
+            return done();
+        }
+        return async.parallel({
+            bucketQuota: parallelDone => {
+                if (bucketQuota > 0) {
+                    return ScubaClientInstance.getUtilizationMetrics('bucket',
+                        `${bucket.getName()}_${creationDate}`, null, {
+                        action: apiName,
+                        inflight,
+                    }, (err, bucketMetrics) => {
+                        if (err) {
+                            return parallelDone(err);
+                        }
+                        if (bucketMetrics.bytesTotal > bucketQuota) {
```

Comment:
we should check here as well that the data returned by Scuba is not stale (as the quota request will return the date as well)

Reply:
The logic was updated with the `isMetricStale` method